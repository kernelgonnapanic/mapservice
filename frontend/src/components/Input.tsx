import React from 'react'
import { FieldProps } from 'formik'
import { TextField } from '@material-ui/core'
import { TextFieldProps } from 'material-ui'

interface Props extends FieldProps, TextFieldProps {
	label: string
}

const Input: React.FC<Props> = ({ placeholder, field, label }) => {
	return (
		<TextField
			variant="outlined"
			label={label}
			placeholder={placeholder}
			{...field}
		/>
	)
}

export default Input
