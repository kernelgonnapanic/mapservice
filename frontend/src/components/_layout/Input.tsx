import React from 'react'
import styled from 'styled-components'
import { FieldProps } from 'formik'
import { TextField } from '@material-ui/core'
import { TextFieldProps } from 'material-ui'

interface Props extends FieldProps, TextFieldProps {
	label: string
	error: boolean
}

const Input: React.FC<Props> = ({
	field,
	form: { touched, errors },
	label,
	type = 'text',
}) => {
	const { name } = field

	return (
		<>
			<TextField
				type={type}
				variant="outlined"
				label={label}
				{...field}
				error={touched[name] && errors[name] ? true : false}
				helperText={touched[name] && errors[name] && errors[name]}
			/>
		</>
	)
}

export default Input
