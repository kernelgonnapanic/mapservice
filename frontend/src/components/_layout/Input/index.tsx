import React from 'react'

interface Props {
	onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
	value: string
	name: string
	placeholder: string
}

const Input: React.FunctionComponent<Props> = ({
	onChange,
	value,
	name,
	placeholder,
}) => {
	return (
		<input
			onChange={onChange}
			value={value}
			name={name}
			placeholder={placeholder}
		></input>
	)
}

export default Input
