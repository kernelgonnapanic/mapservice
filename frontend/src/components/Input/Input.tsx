import React from 'react'
import { FieldProps } from 'formik'
import * as S from './Input.styles'
import { TextFieldProps } from 'material-ui'

interface Props extends FieldProps, TextFieldProps {
	label: string
	error: boolean
	multiline: boolean
	rows: number
}

const Input: React.FC<Props> = ({
	field,
	form: { touched, errors },
	label,
	multiline,
	rows = 0,
	type = 'text',
}) => {
	const { name } = field


	return (
		<>
			<S.Input
				type={type}
				variant="outlined"
				label={label}
				multiline={multiline}
				rows={rows}
				{...field}
				error={touched[name] && errors[name] ? true : false}
				helperText={touched[name] && errors[name] && errors[name]}
			/>
		</>
	)
}

export default Input
