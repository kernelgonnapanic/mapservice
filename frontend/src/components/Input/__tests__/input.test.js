import React from 'react'
import ReactDOM from 'react-dom'
import {render, cleanup} from '@testing-library/react'
import Input from "../Input";
// import {theme} from "../../../App";
// import {ThemeProvider} from "styled-components";

afterEach(cleanup);

test("render Input component", async () => {
    const {asFragment} = render(<Input label="test" error={null} multiline={false} rows={null} field={null} form={null} meta={null}/>)
    expect(asFragment()).toMatchSnapshot()
})

test("render Input Error corretly component", async () => {
    // const container = document.createElement('div');
    //
    // ReactDOM.render(<Input label="test" error={null} multiline={false} rows={null} field={null} form={null} meta={null}/>, container)
    //
    // expect()

})