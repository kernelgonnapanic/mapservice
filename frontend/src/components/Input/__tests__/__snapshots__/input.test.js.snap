// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Input component 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root sc-AxjAm fEWskf"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
      data-shrink="false"
    >
      test
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input"
        type="text"
        value=""
      />
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-59 MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="PrivateNotchedOutline-legendLabelled-61"
        >
          <span>
            test
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;
